### PR TITLE
Move to saz limits

### PR DESCRIPTION
**Pull request by deanwilson**
In an attempt to move to mature, 3rd party, modules this PR covers a possible migration from harden::limits to saz/limits. It should produce the same settings as using `Harden` but with each value in its own files, and none bunched in to limits.conf

The story is located at https://trello.com/c/nVs0krua

### CODE REVIEW

**Comment by rjw1** [comment]
```
==> backend-2.backend: Error: Cannot alias File[/etc/security/limits.d/] to ["/etc/security/limits.d"] at /var/govuk/govuk-puppet/vendor/modules/limits/manifests/init.pp:12; resource ["File", "/etc/security/limits.d"] already declared at /var/govuk/govuk-puppet/vendor/modules/harden/manifests/init.pp:54
```
when run under vagrant

File: Puppetfile [hunk 1]
```diff
@@ -49,7 +50,7 @@ mod 'gdsoperations/gor',
   :ref => 'v0.2.0'
 mod 'alphagov/harden',
   :git => 'git://github.com/alphagov/puppet-harden.git',
-  :ref => 'fa2c51e994b87e71e7b424b6c0048b70c2df36b0'
+  :ref => '69bcbb3'
```
Comment (mattbostock) [inline Puppetfile]: We should probably cut a new release of puppet-harden and use that.

**Comment by mattbostock** [comment]
This PR should also remove the old copy of the vendored puppet-harden module.

**Comment by mattbostock** [comment]
@deanwilson It'd be good to say in your commit messages and/or PR description if this change makes any functional difference; are any new limits set which weren't previously?

**Comment by deanwilson** [comment]
I've updated the PR to mention the functional side. Nothing should really change but the harden module was inconsistent in how it set things so there will be a minor change as settings break out in to individual files to be consistent with all the other usages of `limits`.

**Comment by mattbostock** [comment]
Thanks @deanwilson :+1: It makes it easier for people to test if they know what to look for.

**Comment by deanwilson** [comment]
Must have missed one, which vendored file needs to be removed?

**Comment by mattbostock** [comment]
You can find it using the old SHA from the Puppetfile.lock, it's:
`vendor/puppet/source/fa2c51e994b87e71e7b424b6c0048b70c2df36b0.tar.gz`

**Comment by deanwilson** [comment]
Removed the old file. Thanks.

**Comment by mattbostock** [comment]
LGTM, are you able to test @rjw1's comment in Vagrant?

**Comment by rjw1** [comment]
:+1: lgtm

**Pull request merged**